modules/core/src/main/scala/gem/parser/Misc.scala

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.parser
+
+import atto._, Atto._
+import scala.annotation.tailrec
+
+/** General-purpose parsers and combinators that aren't provided by atto. */
```

Comment:
Why not upstream them? They seem quite useful for any kind of parsing in general.